crates/gitbutler-branch-actions/tests/virtual_branches/workspace_migration.rs

```diff
@@ -0,0 +1,31 @@
+use gitbutler_branch::VirtualBranchesHandle;
+use gitbutler_branch_actions::{update_workspace_commit, verify_branch};
+use gitbutler_operating_modes::{INTEGRATION_BRANCH_REF, WORKSPACE_BRANCH_REF};
+
+/// Tests that "verify branch" won't complain if we are on the old integration
+/// branch, and that `update_workspace_commit` will put us back on the a branch
+/// with the new name.
+#[test]
+fn works_on_integration_branch() -> anyhow::Result<()> {
+    let ctx = gitbutler_testsupport::read_only::fixture(
+        "for-workspace-migration.sh",
+        "workspace-migration",
+    )?;
+    let mut guard = ctx.project().exclusive_worktree_access();
+    let perm = guard.write_permission();
+
+    // Check that we are on the old `gitbutler/integration` branch.
+    assert_eq!(
+        ctx.repository().head()?.name(),
+        Some(INTEGRATION_BRANCH_REF)
+    );
+
+    // Should not throw verification error until migration is complete.
+    let result = verify_branch(&ctx, perm);
+    assert!(result.is_ok());
+
+    // Updating workspace commit should put us on the workspace branch.
+    update_workspace_commit(&VirtualBranchesHandle::new(ctx.project().gb_dir()), &ctx)?;
+    assert_eq!(ctx.repository().head()?.name(), Some(WORKSPACE_BRANCH_REF));
+    Ok(())
+}
```

Comment:
>Where does the code which actually moves you over live, and I would love to see some tests which exercised the code path before we merge in.

It's a call to update_workspace_commit that actually moves you over, so what we need to prevent is `verify_branch` throwing an error while you're still on `gitbutler/integration`. Can you think of additional steps we should be taking?

Reply:
@mtsgrd you should call `assure_open_workspace_mode` to only do the move whilst on `gitbutler/integration`

Reply:
@Caleb-T-Owens that function does not have any side-effects and makes the test fail. What did you have in mind?

<img width="529" alt="image" src="https://github.com/user-attachments/assets/9b54743e-e543-4513-b8ee-3df171cf29a8">

Reply:
Done.

Reply:
Btw, we should probably use the verb _ensure_ rather than _assure_. Ensure means “make certain that something will occur.” Assure means “convince” or “give confidence to.”